lxd/instance/drivers/driver_qemu.go

```diff
@@ -3295,29 +3295,45 @@ func (d *qemu) addGPUDevConfig(sb *strings.Builder, bus *qemuBus, gpuConfig []de
 }
 
 func (d *qemu) addUSBDeviceConfig(sb *strings.Builder, bus *qemuBus, usbConfig []deviceConfig.RunConfigItem) error {
+	var index int
+
+	addOne := func(devName, hostDevice string) error {
+		tplFields := map[string]interface{}{
+			"hostDevice": hostDevice,
+			"devName":    fmt.Sprintf("%s-%d", devName, index),
+		}
+
+		err := qemuUSBDev.Execute(sb, tplFields)
+		if err != nil {
+			return err
+		}
+
+		// Add path to external devPaths. This way, the path will be included in the apparmor profile.
+		d.devPaths = append(d.devPaths, hostDevice)
+		index++
+
+		return nil
+	}
+
 	var devName, hostDevice string
 
 	for _, usbItem := range usbConfig {
```

Comment:
I wouldn't have expected this `usbConfig` to ever contain more than 2 items, "devName" and "hostDevice".

This actually seems more like a bug in the `usb` device code itself whereby its not failing if the selection criteria match multiple USB devices:

https://github.com/lxc/lxd/blob/master/lxd/device/usb.go#L133-L189

Other similar device types have a check that prevent the selection criteria matching multiple devices, e.g.

https://github.com/lxc/lxd/blob/master/lxd/device/gpu_mdev.go#L68-L70

Reply:
@stgraber am I right here, or do we support having a single logical LXD `usb` device pass through multiple actual USB devices? Either way we have a bug in the `usb` driver.

Reply:
Yes it does indeed, but I am questioning if that is correct or not.

Reply:
Given that vendor:product is not a unique property *and* that LXD allows matching on just the vendor (with any number of product matches) as a designed feature, I think it is unavoidable.